command/rand/rand.go

```diff
@@ -0,0 +1,231 @@
+package rand
+
+import (
+	"bufio"
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/urfave/cli"
+	"go.step.sm/cli-utils/command"
+	"go.step.sm/cli-utils/errs"
+	"go.step.sm/crypto/fingerprint"
+	"go.step.sm/crypto/randutil"
+)
+
+func init() {
+	cmd := cli.Command{
+		Name:      "rand",
+		Action:    command.ActionFunc(randAction),
+		Usage:     "generate random strings",
+		UsageText: "**step rand** [<length>] [--format=<format>] [--dictionary=<file>]",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "format",
+				Usage: "The <format> of the output string. See help for list of available formats.",
+			},
+			cli.StringFlag{
+				Name:  "dictionary,dict",
+				Usage: "The <file> to use as a dictionary to get random words.",
+			},
+		},
+		Description: `**step rand** generates random strings that can be used for multiple purposes.
+The <rand> command supports printing stings with different formats. It defaults
+to use the printable characters of the ASCII table; <rand> also supports
+generating a memorable password using words from a provided dictionary.
+
+The length of the random string will default to 32 characters or 6 words
+separated by a dash (-) if a dictionary is used.
+
+The list of supported formats is the following:
+
+* ascii: generates a random string using the 94 printable characters of the
+  ASCII table.
+* alphanumeric: uses the 62 alphanumeric characters in the POSIX/C locale
+  (a-z+A-Z+0-9).
+* alphabet: uses the 52 alphabetic characters in the POSIX/C locale (a-z+A-Z).
+* hex: uses the 16 hexadecimal characters in lowercase (0-9+a-f).
+* dec: uses the 10 decimal characters (0-9).
+* lower: uses the 26 lowercase alphabetic characters in the POSIX/C locale
+  (a-z).
+* upper: uses the 26 uppercase alphabetic characters in the POSIX/C locale
+  (A-Z).
+* emoji: uses a curated list of 256 emojis that are not entirely similar.
+* raw: uses random bytes.
+
+The following special formats are also supported:
+
+* dice: generates a random number between 1 and 6 or the given argument,
```

Comment:
Technically, this is a multiple, but it's just a single die being rolled.

Reply:
I kinda agree here - I assumed the command was going to roll multiple dice for me before reading the example.

Reply:
@dopey @hslatman should I change the format name to `die`?

Reply:
Why even have 'die' or 'dice` as a special format? Like, why not just use `step rand int` or `step rand num` and then allow to specify the start and end inclusive or exclusive.

Reply:
It seems like the only reason to use dice is because the number 6 matches the number of default words in the dictionary format - and I don't think that relation makes sense. I already commented about that earlier.

Reply:
`step rand int` is a good addition that can be added later, but it requires more flags. 🎲 is fun. A die of 6 is the most common one. The match between the number of words in the dictionary and the default die is just a coincidence. Six looked like a good default.